src/queue/queue.ts

```diff
@@ -0,0 +1,48 @@
+// This file defines a basic queue data structure, providing
+// the standard Array push(), shift(), and length attributes,
+// but with O(1) operation.
+
+// Private helper class.
+class Cell<T> {
+  public next :Cell<T> = null;
+
+  constructor(private item_:T) {}
+
+  public getItem() {
```

Comment:
There's a weird mix of private and public fields in this file. Generally I'm against public fields but for Cell specifically I think you could make `item_` a public field.